src/providers/realtime-provider.js

```diff
@@ -86,98 +99,172 @@ export default class RealtimeProvider {
         };
     }
 
-    _connect() {
-        this.socket.onopen(this._onopen.bind(this));
-        this.socket.onmessage(this._onmessage.bind(this));
-    }
+    sendSubscribeMessage(subscriptionDetails) {
+        let domainObject = subscriptionDetails.domainObject;
+        let message = MESSAGES.SUBSCRIBE[domainObject.type](subscriptionDetails);
 
-    _isSupportedDataType(type) {
-        return this.supportedDataTypes[type];
+        this.sendOrQueueMessage(message);
     }
 
-    _isSupportedObjectType(type) {
-        return this.supportedObjectTypes[type];
+    sendUnsubscribeMessage(subscriptionDetails) {
+        let message = MESSAGES.UNSUBSCRIBE(subscriptionDetails);
+
+        this.sendOrQueueMessage(message);
     }
 
-    _onmessage(event) {
-        let data = JSON.parse(event.data);
+    reconnect() {
+        this.subscriptionsByCall.clear();
 
-        if (!this._isSupportedDataType(data.type)) {
+        if (this.reconnectTimeout) {
             return;
         }
 
-        let isReply = data.type === DATA_TYPES.DATA_TYPE_REPLY;
-        let subscriptionDetails;
+        this.reconnectTimeout = setTimeout(() => {
+            this.connect();
+            delete this.reconnectTimeout;
+        }, FALLBACK_AND_WAIT_MS[this.currentWaitIndex]);
 
-        if (isReply) {
-            let id = data.data.replyTo;
-            let call = data.call;
-            subscriptionDetails = this.subscriptionsById[id];
-            if (!subscriptionDetails) {
-                return;
-            }
+        if (this.currentWaitIndex < FALLBACK_AND_WAIT_MS.length - 1) {
+            this.currentWaitIndex++;
+        }
+    }
 
-            subscriptionDetails.call = call;
-            this.subscriptionsByCall.set(call, subscriptionDetails);
+    sendOrQueueMessage(request) {
+        if (this.connected) {
+            try {
+                this.sendMessage(request);
+                return true;
+            } catch (error) {
+                this.connected = false;
+                console.error(error);
+                console.warn("Error while attempting to send to websocket. Reconnecting...");
+
+                this.requests.push(request);
+                this.reconnect();
```

Comment:
Do we need to return false here? I see you're returning true above. Is anything relying on this? I didn't see anything, if that's true, you may be able to remove that `return true` above.